shopelectro/tests/tests_models.py

```diff
@@ -21,3 +24,42 @@ def test_creation_deactivated_product(self):
             ))
         except Exception as error:
             self.fail(f'Creation of existing product failed: {{ error }}')
+
+
+class TagModel(TestCase):
+
+    fixtures = ['dump.json']
+
+    def get_products(self):
+        return Product.objects.all()[:3]
+
+    def test_get_brands_content(self):
+        brand_group = TagGroup.objects.get(name=settings.BRAND_TAG_GROUP_NAME)
+        for product, tag in Tag.objects.get_brands(self.get_products()).items():
+            self.assertEquals(tag.group, brand_group)
+            self.assertIn(product, tag.products.all())
+
+    def test_filter_by_products_equality(self):
+        products = self.get_products()
+        self.assertEquals(
+            set(Tag.objects.filter_by_products(products)),
+            set(chain.from_iterable(product.tags.all() for product in products)),
+        )
+
+    def test_filter_by_products_quantity(self):
```

Comment:
equality and quantity both are props of the same structure.
It'll be more correct to test it with one case, i think.

Draft code proposal:
```
sorted_by_id = partial(sorted(key=lambda x: x.id))

self.assertEquals(
            sorted_by_id(list(Tag.objects.filter_by_products(products))),
            sorted_by_id(list(chain.from_iterable(product.tags.all() for product in products))),
        )
```

Decide yourself

Reply:
@duker33 We have to use `set(...)` in this case, because  `chain.from_iterable(product.tags.all() for product in products)` should contain unique tags, therefore we can not use `sorted_by_id(list(..))`